lib/features/login/ui/widgets/email_password.dart

```diff
@@ -0,0 +1,104 @@
+import 'package:doctor_appointment/core/helpers/app_regex.dart';
+import 'package:doctor_appointment/core/widgets/add_text_form_field.dart';
+import 'package:doctor_appointment/features/login/ui/widgets/password_validations.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:gap/gap.dart';
+
+import '../../logic/cubit/login_cubit.dart';
+
+class EmailAndPasswrod extends StatefulWidget {
+  const EmailAndPasswrod({super.key});
+
+  @override
+  State<EmailAndPasswrod> createState() => _EmailAndPasswrodState();
+}
+
+class _EmailAndPasswrodState extends State<EmailAndPasswrod> {
+  bool isObsecureText = true;
+  late TextEditingController passwordController;
+  bool hasLowercase = false;
+  bool hasUppercase = false;
+  bool hasSpecialCharacters = false;
+  bool hasNumber = false;
+  bool hasMinLength = false;
+  @override
+  void initState() {
+    super.initState();
+    passwordController = context.read<LoginCubit>().passwordController;
+    setupPasswordControllerListener();
+  }
+
+  void setupPasswordControllerListener() {
+    passwordController.addListener(() {
+      setState(() {
+        hasLowercase = AppRegex.hasLowerCase(passwordController.text);
+        hasUppercase = AppRegex.hasUpperCase(passwordController.text);
+        hasSpecialCharacters =
+            AppRegex.hasSpecialCharacter(passwordController.text);
+        hasNumber = AppRegex.hasNumber(passwordController.text);
+        hasMinLength = AppRegex.hasMinLength(passwordController.text);
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Form(
+      key: context.read<LoginCubit>().formKey,
+      child: Column(
+        children: [
+          AddTextFormField(
+            labelText: 'Email',
+            suffixIcon: const Icon(Icons.email),
+            hintText: 'Enter Your E-mail',
+            validator: (value) {
+              if (value == null ||
+                  value.isEmpty ||
+                  !AppRegex.isEmailValid(value)) {
+                return 'Please enter a valid E-mail address';
+              }
+            },
+            controller: context.read<LoginCubit>().emailController,
+          ),
+          Gap(20.h),
+          AddTextFormField(
+            validator: (value) {
```

Comment:
can return null

Reply:
try to use const, you can add it by default when saving

Reply:
done

Reply:
added return Null